src/main/scala/dsptools/numbers/binary_types/BinaryRepresentation.scala

```diff
@@ -20,5 +22,8 @@ trait BinaryRepresentation[A <: Data] extends Any {
   def mul2(a: A, n: Int): A = shl(a, n)
   // Trim to n fractional bits (with DspContext) -- doesn't affect DspReal
   def trimBinary(a: A, n: Int): A = trimBinary(a, Some(n))
-  def trimBinary(a: A, n: Option[Int]): A 
+  def trimBinary(a: A, n: Option[Int]): A
+
+  // Clip A to B (range)
+  def clip(a: A, b: A): A = ???
```

Comment:
I think the implementation (`???`) should not be here, but in whatever implements the trait

Reply:
What should the implementation be, or should I just throw an error for now. Making this method abstract requires implementing in DspComplex, DspReal, Interval, FixedPoint, SInt, UInt type classes

Reply:
It seems to me that all the inheritors should all have `???` until implemented so this one can be abstract. I think it's probably not good to have a default implementation for something like this because weird stuff can happen if you forget to override it. However, most of implementations can probably have something like

```
def clip(a: A, b: A): A = {
  requireIsHardware(a)
  requireIsHardware(b)
  WireDefault(b, a)
}
```